jaeger-core/src/test/java/com/uber/jaeger/samplers/PerOperationSamplerTest.java

```diff
@@ -103,21 +103,21 @@ public void testPerOperationSamplerWithKnownOperation() {
 
   @Test
   public void testUpdate(){
-    GuaranteedThroughputSampler guaranteedThroughputSampler = mock(GuaranteedThroughputSampler.class);
+    GuaranteedThroughputSampler guaranteedThroughputSampler = new GuaranteedThroughputSampler(SAMPLING_RATE, DEFAULT_LOWER_BOUND_TRACES_PER_SECOND);
     operationToSamplers.put(OPERATION, guaranteedThroughputSampler);
 
     PerOperationSamplingParameters perOperationSamplingParameters =
-        new PerOperationSamplingParameters(OPERATION, new ProbabilisticSamplingStrategy(SAMPLING_RATE));
+        new PerOperationSamplingParameters(OPERATION, new ProbabilisticSamplingStrategy(0.001));
     List<PerOperationSamplingParameters> parametersList = new ArrayList<>();
     parametersList.add(perOperationSamplingParameters);
 
     OperationSamplingParameters parameters =
         new OperationSamplingParameters(DEFAULT_SAMPLING_PROBABILITY,
-                                        DEFAULT_LOWER_BOUND_TRACES_PER_SECOND, parametersList);
+                                        1.0, parametersList);
 
     assertTrue(undertest.update(parameters));
-    //Checks that creating a new instance with the given parameters is the same as updating an existing instance
-    assertEquals(new PerOperationSampler(MAX_OPERATIONS, parameters), undertest);
+    assertEquals(new GuaranteedThroughputSampler(0.001, 1.0),
```

Comment:
This test doesn't really test whether the `guaranteedThroughputSampler` is a new instance, or one that was updated by using the `update` method. 
I suggest using a mock for the `guaranteedThroughputSampler` and verifying that update was called with the right parameters.